atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -22,23 +22,22 @@
 import com.google.common.base.Preconditions;
 import com.palantir.atlasdb.transaction.api.Transaction;
 import com.palantir.atlasdb.transaction.api.TransactionFailedException;
-import com.palantir.atlasdb.transaction.api.TransactionFailedRetriableException;
 import com.palantir.atlasdb.transaction.api.TransactionManager;
 import com.palantir.atlasdb.transaction.api.TransactionTask;
 import com.palantir.common.base.Throwables;
 
 public abstract class AbstractTransactionManager implements TransactionManager {
     public static final Logger log = LoggerFactory.getLogger(AbstractTransactionManager.class);
 
-    private volatile boolean isClosed = false;
+    private volatile boolean closed = false;
 
     @Override
     public <T, E extends Exception> T runTaskWithRetry(TransactionTask<T, E> task) throws E {
-        checkOpen();
         int failureCount = 0;
         while (true) {
+            checkOpen();
             try {
-                return internalRunTaskThrowOnConflict(task);
+                return runTaskThrowOnConflict(task);
```

Comment:
@SerialVelocity why did you revert this? This ensures that the txn manager is open before running before delegating to the concrete `runTaskThrowOnConflict` impl